app/src/main/java/com/omkarmoghe/pokemap/controllers/map/directions/DirectionsManager.java

```diff
@@ -0,0 +1,255 @@
+                            }
+                        }
+
+                        @Override
+                        public void onDirectionFailure(Throwable t) {
+                            // t.getCause() or t.getLocalizedMessage() or t.getMessage() ?
+                            notifyRouteOptionsFailed(t.getLocalizedMessage());
+                        }
+                    });
+
+            // GET BICYCLING INFO
+            GoogleDirection.withServerKey(mPref.getDirectionsAPIKey())
+                    .from(from)
+                    .to(to)
+                    .transportMode(TransportMode.BICYCLING)
+                    .unit(Unit.METRIC)
+                    .language(Locale.getDefault().getLanguage())
+                    .execute(new DirectionCallback() {
+                        @Override
+                        public void onDirectionSuccess(Direction direction, String rawBody) {
+                            if (direction.isOK()) {
+                                // No alternative routes asked so only 1 Route in list
+                                // No waypoints set so only one Leg in list
+                                Leg leg = direction.getRouteList().get(0).getLegList().get(0);
+                                ArrayList<LatLng> directionPositionList = leg.getDirectionPoint();
+
+                                PolylineOptions polylineOptions = DirectionConverter.createPolyline(context, directionPositionList, 5, lineColor);
+
+                                int durationSec = Integer.parseInt(leg.getDuration().getValue());
+                                float durationMin = (float) durationSec / 60;
+
+                                options.setBikeOption(new RouteOption(
+                                        from,
+                                        to,
+                                        TransportMode.WALKING,
+                                        leg.getDistance().getValue(),
+                                        durationFormat.format(durationMin),
+                                        polylineOptions));
+                                if (options.isComplete()) {
+                                    notifyRouteOptionsFound(options);
+                                }
+                            } else {
+                                // getErrorMessage() or getStatus() ?
+                                notifyRouteOptionsFailed(direction.getErrorMessage());
+                            }
+                        }
+
+                        @Override
+                        public void onDirectionFailure(Throwable t) {
+                            // t.getCause() or t.getLocalizedMessage() or t.getMessage() ?
+                            notifyRouteOptionsFailed(t.getLocalizedMessage());
+                        }
+                    });
+
+            // GET DRIVING INFO
+            GoogleDirection.withServerKey(mPref.getDirectionsAPIKey())
```

Comment:
do we need to request all 3 transport modes? or maybe refactor these methods into one.